tests/generation/test_utils.py

```diff
@@ -2438,28 +2369,40 @@ def _check_encoder_hidden_states_for_generate(self, hidden_states, batch_size, c
         )
 
     def _check_past_key_values_for_generate(self, batch_size, past_key_values, seq_length, config, num_beam_groups=1):
-        self.assertIsInstance(past_key_values, tuple)
-        self.assertListEqual(
-            [isinstance(iter_past_key_values, tuple) for iter_past_key_values in past_key_values],
-            [True] * len(past_key_values),
-        )
-
+        self.assertIsInstance(past_key_values, (tuple, Cache))
```

Comment:
This PR makes `generate` return a `Cache` object by default -- `Cache` checks were added